### ISSUE TITLE
Dispatch sign up form with additional fields

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6897]
# Story

As a Facilities website owner, I would like the Dispatch sign up form on SiteNow to contain all fields that are part of our Dispatch form.

# Additional Information

Here is a link to sign up to this subscription list through Dispatch https://apps.its.uiowa.edu/dispatch/subscriptionLists/574308527/signup

Here is what outputs using the SiteNow Dispatch Sign up Form https://facilities.prod.drupal.uiowa.edu/alerts

![Screenshot 2023-09-18 at 3 27 57 PM](https://github.com/uiowa/uiowa/assets/1017022/62271222-7ed2-4ac8-ba2f-f0db1aa89ed0)

They would like to include phone and Unit/Organization in the SiteNow feature as well.

I believe Unit/Organization is a custom field. We'll have to see if those custom fields exist in the API for us to use.

If we can't pull in custom fields, I think we could iframe this but let's see what's possible first.

# Proposed solution

**Comment by joewhitsitt** [comment]
>We'll have to see if those custom fields exist in the API for us to use.

Doesn't look like the custom field data is available through the API currently. https://apps.its.uiowa.edu/dispatch/api-ref#/Populations/get_populations__id_

<img width="511" alt="Screenshot 2023-10-10 at 11 37 07 AM" src="https://github.com/uiowa/uiowa/assets/4663676/50d745d3-9ed5-4484-bb38-b03d78ca2d5b">

**Comment by joewhitsitt** [comment]
Reached out to Michael A with its-web cc'd to see if this is possible.

**Comment by briand44** [comment]
Sounded like this may be available in the api in a week or so? Going to change priority a bit here in the hope if we delay this, it will be available soon.

**Comment by joewhitsitt** [comment]
Sounds like this is being worked on. Michael shared a sneak peek:

```
"id": "16900",
  "name": "Subscription List",
  "description": "My super descriptive description.",
  "dataSourceType": "SubscriptionList",
  "tags": [
    ""
  ],
  "parameterized": false,
  "archived": false,
  "hidden": false,
  "subscriptionList": {
    "optInType": "Double",
    "listOwnerEmail": "dispatch-support@uiowa.edu",
    "subscribeCallbackUrl": "http://localhost/dispatch/test",
    "displayName": "My Subscription List",
    "displayDescription": "My very vague description.",
    "credentialId": null,
    "customFields": [
      {
        "fieldType": "TEXT",
        "key": "program",
        "label": "Program",
        "listOptions": "",
        "required": false,
        "defaultValue": "",
        "helpText": "",
        "sortOrder": 0
      },
      {
        "fieldType": "DROPDOWN",
        "key": "thingone",
        "label": "One",
        "listOptions": "alpha\r\nbeta\r\ngamma",
        "required": false,
        "defaultValue": "beta",
        "helpText": "The thing with the stuff",
        "sortOrder": 1
      },
      {
        "fieldType": "RADIO",
        "key": "thingtwo",
        "label": "Two",
        "listOptions": "Red\r\nOrange\r\nYellow\r\nGreen\r\nBlue\r\nIndigo\r\nViolet",
        "required": false,
        "defaultValue": "Blue",
        "helpText": "Please answer this question.",
        "sortOrder": 2
      }
    ]
  },
  "client": "/dispatch/api/v1/client"
}
```

**Comment by joewhitsitt** [comment]
>The phone field is standard and not required. Giving people the option to toggle its display is on a wish list somewhere.

I told Michael we may build that toggle on to our side for the short term but to let us know if it is possible on the dispatch side.

**Comment by briand44** [comment]
Phone toggle is handled on Dispatch side. Evaluate using test endpoint and let them know when they can deploy to prod API

**Comment by joewhitsitt** [comment]
<img width="182" alt="Screenshot 2023-10-23 at 4 10 12 PM" src="https://github.com/uiowa/uiowa/assets/4663676/5b50c639-4e97-4c64-9061-d9f55e2312ab">

Looks like there might be more types to account for

**Comment by cory-skeers** [comment]
Resolved by #7032
